From db02cef24a1e8f737036c7ab052260dd81fed1ba Mon Sep 17 00:00:00 2001
From: Miaoqing Pan <quic_miaoqing@quicinc.com>
Date: Mon, 10 Feb 2025 10:14:14 +0800
Subject: [PATCH] PENDING: data: disable systemd-networkd for confliction

NetworkManager conflicts with systemd-netword. Disable systemd-networkd
when NetworkManager is running.

Signed-off-by: Miaoqing Pan <quic_miaoqing@quicinc.com>
Upstream-Status: Pending
---
 data/NetworkManager.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/NetworkManager.service.in b/data/NetworkManager.service.in
index f09ae86..272c09f 100644
--- a/data/NetworkManager.service.in
+++ b/data/NetworkManager.service.in
@@ -11,6 +11,7 @@ Type=dbus
 BusName=org.freedesktop.NetworkManager
 ExecReload=/usr/bin/busctl call org.freedesktop.NetworkManager /org/freedesktop/NetworkManager org.freedesktop.NetworkManager Reload u 0
 #ExecReload=/bin/kill -HUP $MAINPID
+ExecStartPre=/bin/sh -c 'systemctl stop systemd-networkd.socket && systemctl disable systemd-networkd.socket && systemctl stop systemd-networkd && systemctl disable systemd-networkd'
 ExecStart=@sbindir@/NetworkManager --no-daemon
 Restart=on-failure
 # NM doesn't want systemd to kill its children for it
-- 
2.25.1

